src/pip/_internal/metadata/importlib/_compat.py

```diff
@@ -2,6 +2,15 @@
 from typing import Any, Optional, Protocol, cast
 
 
+class BadMetadata(ValueError):
```

Comment:
This should also derive from PipError?

Reply:
Not sure what the convention is. This is supposed to be somewhat internal to the `metadata` subpackage and shouldn’t be uncaught.